src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunHelperNode.cs

```diff
@@ -40,6 +40,25 @@ public ReadyToRunHelperNode(ReadyToRunHelperId id, Object target)
         {
             _id = id;
             _target = target;
+
+            switch (id)
+            {
+                case ReadyToRunHelperId.NewHelper:
+                case ReadyToRunHelperId.NewArr1:
+                case ReadyToRunHelperId.IsInstanceOf:
+                case ReadyToRunHelperId.CastClass:
+                case ReadyToRunHelperId.GetNonGCStaticBase:
+                case ReadyToRunHelperId.GetGCStaticBase:
+                case ReadyToRunHelperId.GetThreadStaticBase:
+                    {
+                        // Make sure that if the EEType can't be generated, we throw the exception now.
+                        // This way we can fail generating code for the method that references the EEType
+                        // and (depending on the policy), we could avoid scraping the entire compilation.
+                        TypeDesc type = (TypeDesc)target;
+                        EETypeNode.CheckCanGenerateEEType(type);
```

Comment:
Maybe actually grabbing the EETypeNode here from the factory might be better (same "caching of answers" argument as above).